.github/e2e/e2e_test.go

```diff
@@ -0,0 +1,305 @@
+	tasclient "github.com/intel/telemetry-aware-scheduling/telemetry-aware-scheduling/pkg/telemetrypolicy/client/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+
+	"path/filepath"
+)
+
+/*
+Metrics values are currently set with a mount file which is then read by the node exporter. This behaviour could be
+changed in future to allow the setting of metrics natively inside the e2e testing code. For this first iteration a new
+metric and policy will be used for each of the three e2e smoke tests being reviewed.
+
+*/
+
+var (
+	kubeConfigPath *string
+	cl             *kubernetes.Clientset
+	tascl          *tasclient.Client
+	cm             metrics.CustomMetricsClient
+)
+
+//init sets up the clients used for the end to end tests
+func init() {
+	if home := homedir.HomeDir(); home != "" {
+		kubeConfigPath = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "path to your kubeconfig file")
+	} else {
+		kubeConfigPath = flag.String("kubeconfig", "", "require absolute path to your kubeconfig file")
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeConfigPath)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// create the clientset
+	cl, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	cm = metrics.NewClient(config)
+
+	tascl, err = tasclient.New(*config, "default")
+	if err != nil {
+		panic(err.Error())
+	}
+	//TODO: Replace the generic timeout with an explicit check for the custom metrics from the API Server which times out after some period
+	err = waitForMetrics(120 * time.Second)
+
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+var (
+	prioritize1Policy = getTASPolicy("prioritize1", scheduleonmetric.StrategyType, "prioritize1_metric", "GreaterThan", 0)
+	filter1Policy     = getTASPolicy("filter1", dontschedule.StrategyType, "filter1_metric", "LessThan", 20)
+	filter2Policy     = getTASPolicy("filter2", dontschedule.StrategyType, "filter2_metric", "Equals", 0)
+	deschedule1Policy = getTASPolicy("deschedule1", deschedule.StrategyType, "deschedule1_metric", "GreaterThan", 8)
+)
+
+func TestTASDeschedule(t *testing.T) {
+	tests := map[string]struct {
+		policy *api.TASPolicy
+		want   map[string]bool
+	}{
+		"Label node for deschedule": {policy: deschedule1Policy, want: map[string]bool{"kind-worker2": true}},
+		//"Unlabel node on policy deletion":    {policy: deschedule1_policy, pod: podForPolicy(fmt.Sprintf("pod-%v", rand.String(8)), deschedule1_policy.Name), want:  map[string]bool{"kind-worker2":false}},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			res := map[string]bool{}
+			log.Printf("Running: %v\n", name)
+			//defer the running of a cleanup function to remove the policy and pod after the test case
```

Comment:
Perhaps remove here and add as comment in cleanup function?

Reply:
I think we should keep this - I want to be really clear on what this means for the flow of the test to make it easier to change down the line.